src/rt/util/array.d

```diff
@@ -63,3 +63,9 @@ private void _enforceNoOverlap(in char[] action,
     msg ~= bytes.unsignedToTempString(tmpBuff, 10);
     throw new Error(msg);
 }
+
+private const(void)* arrayToPtr(const void[] array) @trusted
+{
+    // Ok because the user will never dereference the pointer
```

Comment:
Not ok...  If safe code can call this method and thereby break memory safety, it should not be trusted.  The fact that the callers above don't do this is just a convention - they are effectively `@trusted` thanks to this function.